### ISSUE TITLE
[java] AvoidFieldNameMatchingMethodName always reported twice

### ISSUE THREAD

**Issue by Ledmington** [pmd/pmd#5570]
**Affects PMD Version:** 
7.11.0

**Description:**
If you have a field and a method in the same class with the same name, the rule `AvoidFieldNameMatchingMethodName` is rightfully triggered but always twice and always at the line of the field's declaration. Moreover, if you overload the method by adding declarations with different signatures the rule is always triggered/reported twice, never more. I think it would make sense to report it for every method with the same name of a field.

**Code Sample demonstrating the issue:**

```java
public class MyClass {
  private int example;

  public void example(int x) {
    this.example = x;
  }
}
```

**Running PMD through:** *Gradle*

**Comment by oowekyala** [comment]
I cannot reproduce this unfortunately, even with gradle. XPath rules deduplicate their results so even if the xpath query was returning several times the same node, it would be reported once. Therefore if you have a duplicate warning, and you run gradle only once, that must be because there are several instances of that rule in your ruleset. Is it possible that you have the rule twice in your ruleset, maybe with different names/ imported from different XML? Can you share more about your gradle configuration and rulesets? Thank you

**Comment by Ledmington** [comment]
Here is how to reproduce it.

1. Have a little Gradle project, one java file is enough.
2. Have the java file look like this and add as many methods named `example` as you want in the same class:
```java
public class MyClass {
  private int example;

  public void example(int x) {
    this.example = x;
  }
}
```
3. Have the `build.gradle` file look like this:
```groovy
plugins {
    id 'application'
    id 'pmd'
}

repositories {
    mavenCentral()
}

pmd {
    toolVersion = "7.11.0"
    ignoreFailures = false
    consoleOutput = true
    rulesMinimumPriority = 5
    threads = 1
    incrementalAnalysis = true
}

pmdMain {
    ruleSetConfig = resources.text.fromFile("${project.rootDir}/prod.ruleset.xml")
}
```
4. Have the `prod.ruleset.xml` file look like this:
```xml
<?xml version="1.0"?>
<ruleset name="prod-ruleset"
		xmlns="http://pmd.sourceforge.net/ruleset/2.0.0"
		xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
		xsi:schemaLocation="http://pmd.sourceforge.net/ruleset/2.0.0 https://pmd.sourceforge.io/ruleset_2_0_0.xsd">
    <description>Ruleset for production code</description>

    <rule ref="category/java/errorprone.xml">
        <!-- Avoid reporting System.exit() -->
        <exclude name="DoNotTerminateVM" />
    </rule>
</ruleset>
```
5. Run `./gradlew build` or `./gradlew pmdMain`. The console should look like this:
```
...
/app/src/main/java/org/example/MyClass.java:4:        AvoidFieldNameMatchingMethodName:       Field example has the same name as a method
/app/src/main/java/org/example/MyClass.java:4:        AvoidFieldNameMatchingMethodName:       Field example has the same name as a method
...
```

No matter how many methods with the same name are added, the rule `AvoidFieldNameMatchingMethodName` is always triggered and reported twice and both times on the field declaration.

When preparing this example, I discovered that, if I remove the only `rule` block from the `prod.ruleset.xml` file, the `AvoidFieldNameMatchingMethodName` rule gets reported only once.

Versions:
gradle: 8.13
java: OpenJDK 21.0.6+7 Temurin
PMD: 7.11.0

**Comment by adangel** [comment]
Hi @Ledmington ,

in the past (not sure if that is still the case), gradle provides default configurations for the used rules/rulesets and provides **two** properties to configure them. And sometimes all are used.

You configured only the property **ruleSetConfig**, but there is also the property **ruleSets**. And this contains by default the entire category "errorprone": https://docs.gradle.org/current/dsl/org.gradle.api.plugins.quality.PmdExtension.html (conventional value)

Although it is documented differently (if ruleSetConfig is provided then ruleSets shouldn't contain the default errorprone), it seems, this simply doesn't work as documented.

I suggest:

1. try to use property **ruleSets** as documented in the example here https://docs.gradle.org/current/userguide/pmd_plugin.html#sec:pmd_configuration - that way you definitively overwrite this
2. if 1. is not possible, additional set `ruleSets = []` to an empty list
3. report a bug at https://github.com/gradle/gradle/issues
  * Alternatively try to reopen https://github.com/gradle/gradle/issues/8514 - this is IMHO exactly the problem
4. report back the findings here, so that we can either close this issue or verify that we don't have a bug on PMD' site.

**Comment by Ledmington** [comment]
I confirm that adding `ruleSets = []` fixes the issue, as written in PMD docs, and that's what I will do.

**Comment by adangel** [comment]
I'm giving it another try upstream: https://github.com/gradle/gradle/issues/32980 and https://github.com/gradle/gradle/pull/32981
